More async_trait conversions

e00E opened the pull request:
Diff is cleaner without whitespace changes.

### Test Plan
No logic change. CI.

e00E committed:
More async_trait conversions

nlordell commented on services-core/src/driver/stablex_driver.rs:
```diff
@@ -524,12 +516,12 @@ mod tests {
             .with(eq(batch), always(), eq(U256::from(42)), always())
             .returning(|_, _, _, _| {
-                async { Err(SolutionSubmissionError::Benign("Benign Error".to_owned())) }.boxed()
+                Err(SolutionSubmissionError::Benign("Benign Error".to_owned()))
```
This is very nice!

nlordell reviewed (approved):
Approved assuming that there are no logic changes and its just about removing the `async move { ... }.boxed()` around the implementation.

This is a very welcome change!

e00E committed:
Convert TokenInfoFetching to async_trait

A welcome side effect of the conversion is that the mockall workaround
for lifetimes is no longer needed.
In this commit I also had to change one test slightly and move the the
default implementation of find_token_by_address into its own function.
Otherwise we cannot test it as mockall overrides the default
implementation.

e00E committed:
Merge branch 'master' into async-trait-1
